benchmarks/benchmark-micronaut-data-jdbc/build.gradle

```diff
@@ -10,4 +10,5 @@ dependencies {
     implementation "io.micronaut.sql:micronaut-jdbc-hikari"
     runtimeOnly "com.h2database:h2"
     runtimeOnly mn.snakeyaml
+    runtimeOnly mnSerde.micronaut.serde.support
```

Comment:
Now that we changed from codecs to serde and have compileOnly dependency in data-jdbc, all projects using jdbc might need to import serde? I had dependency as implementation, but that caused issue for test resources.

Reply:
We will need to resolve this later, probably because of this I have added `implementation` dependency for serde support in data-jdbc and data-r2dbc but that caused issue for test containers. ObjectMapper might not be always needed/instantiated but it needs to be on the classpath in runtime.

Reply:
The user should always be able to select between Micronaut Serde and Jackson. It's not test-resources per se which fails: it's the fact that there are tests which _do not_ use serde but jackson instead, and those tests are now failing. So data JDBC should be agnostic of the serialization framework. So it's fine if it's a `compileOnly` dependency, and that tests of Micronaut Data pick up serde, but modules themselves shouldn't.

Reply:
I see. Just wanted to say that we changed `DefaultJdbcRepositoryOperations` and added this parameter `@Nullable io.micronaut.serde.ObjectMapper objectMapper` which was not there before and we need to add to the docs so users can update their projects with serde import like this benchmark project.

Reply:
mmm, but doesn't this mean that serde _is_ required at runtime now?

Reply:
Well yes, we wanted to move away from MediaTypeCodec to serde and that is how I have implemented it. That is why I added `implementation` dependency. Maybe we need to revert this now?

Reply:
I guess so, because this breaks the assumption that the user can pick whatever serialization framework they want. But I may be misunderstanding.

Reply:
Yes, maybe should wait and see with Graeme and Denis on Monday what we want to do about this.